cmd/nvidia-mig-manager/main.go

```diff
@@ -298,6 +306,44 @@ func start(c *cli.Context) error {
 	}
 }
 
+// setupRequiredForCDI discovers the paths to libnvidia-ml.so.1 and nvidia-smi
+// when required.
+//
+// After applying a MIG configuration but before generating a CDI spec,
+// it is required to run nvidia-smi to create the nvidia-cap* device nodes.
+// If driverRoot != devRoot, we must discover the paths to libnvidia-ml.so.1 and
+// nvidia-smi in order to run nvidia-smi. We discover the paths here once and
+// pass these as arguments to reconfigure-mig.sh
+//
+// Currently, driverRoot != devRoot only when devRoot='/'. Since mig-manager
+// has rw access to the host rootFS (at hostRootMountFlag), reconfigure-mig.sh
+// will first chroot into the host rootFS before invoking nvidia-smi, so the
+// device nodes get created at '/dev' on the host.
+func setupRequiredForCDI() error {
```

Comment:
This name seems "off" we're not setting anythinng up.

Could we change this to:
```suggestion
func getPathsForCDI() (string, string, error) {
```
I would even go so far as to say that we want: `getDriverPaths() (string, string, error)` that actually contains the path logic and we can call this from `getPathsForCDI`.

Reply:
Agreed. Updated to `getPathsForCDI`.